frame/staking/src/lib.rs

```diff
@@ -1802,8 +1802,8 @@ decl_module! {
 		///   NOTE: weights are assuming that payouts are made to alive stash account (Staked).
 		///   Paying even a dead controller is cheaper weight-wise. We don't do any refunds here.
 		/// # </weight>
-		#[weight = T::WeightInfo::payout_stakers_alive_staked(T::MaxNominatorRewardedPerValidator::get())]
-		fn payout_stakers(origin, validator_stash: T::AccountId, era: EraIndex) -> DispatchResult {
+		#[weight = T::WeightInfo::payout_stakers_alive_staked(T::MaxNominatorRewardedPerValidator::get() + 1)]
```

Comment:
Why is there a (+1) here?

Reply:
Because int the worst case we do payout ops to max number of clipped nominators + 1 validator.

Reply:
yeah this was an oversight in the previous benchmark apparently.

Reply:
No actually you are wrong @emostov 

If you look at the benchmark: https://github.com/paritytech/substrate/blob/master/frame/staking/src/benchmarking.rs#L388

We always assume the validator is paid out. This is baked into the whole benchmark. So even if `MaxNominatorRewardedPerValidator` was zero, then we would still have the correct weight for paying the validator.

Reply:
Yup, I see what you saying - will correct it shortly.

Re-iterating specifics just to make sure I am reading correctly: nominators are bound by `n` [here](https://github.com/paritytech/substrate/blob/0f6b5734e8ff52199987b496b2ef7bfd51655198/frame/staking/src/benchmarking.rs#L85-L86), and we set up the validator [here regardless](https://github.com/paritytech/substrate/blob/0f6b5734e8ff52199987b496b2ef7bfd51655198/frame/staking/src/benchmarking.rs#L65-L70); so the benchmark and resulting analysis used to impl `WeighInfo` will implicitly always count a payout to the validator.

Reply:
Yeah. And I think there really isnt a scenario we dont pay out the validator when caling payout.

Reply:
fwiw there is this [non-error scenario](https://github.com/paritytech/substrate/blob/0f6b5734e8ff52199987b496b2ef7bfd51655198/frame/staking/src/lib.rs#L2010-L2011) where no payouts happen; but given the heuristic to air on the upper-bound of weight to prevent DoS attacks it seems fine.